source/game.programme.adcontract.bmx

```diff
@@ -2004,10 +2004,11 @@ Type TAdContract Extends TBroadcastMaterialSource {_exposeToLua="selected"}
 		If IsLimitedToTargetGroup(TVTTargetGroup.MEN) Then result:* 2
 		'aim at one spot per day left
 		If Self.getDaysLeft() < Self.getSpotsToSend() Then result:* 2
+		If Self.getSpotsToSend() = 1 Then result:* 2
 		'big effort on last day
 		If Self.getDaysLeft() = 0 And GetWorldTime().getHour() > 5 Then result:* 4
 		'the more spots sent the worse failing will be (other spots wasted)
-		If GetSpotsToSendPercentage() < 1 Then result:/ (1-GetSpotsToSendPercentage())
+		'If GetSpotsToSendPercentage() < 1 Then result:/ (1-GetSpotsToSendPercentage())
```

Comment:
Warum hast Du das rausgenommen? Sollte dann nicht stattdessen die "Penalty" einen gewissen Einfluss haben (die Hoehe der Strafe)
Sprich 2 Vertraege, jeweils 1 Spot uebrig ... was bringt mehr - bzw bestraft eines "mehr" ?

Reply:
Ja das stimmt. Ich hatte angenommen, dass die Änderung darüber den Fall mit abdecken würde.
Folgende Konstellation ist problematisch: die KI hat zwei Verträge mit 4 Spots und ähnlichen Anforderungen. Es ist kurz vor Mitternacht und es kann nur einer der beiden Verträge erfüllt werden (nur noch 5 Blöcke zu senden). D.h. es muss verhindert werden, dass die KI immer wieder zwischen den Verträgen wechselt.
Aber das sollte passen; denn der Vertrag mit mehr erfüllten Spots wird bevorzugt.

Reply:
naja ... selbst dann ist es noch nicht "eindeutig".

Wenn zwei Werbevertraege mit gleichen "Modalitaeten" und jeweils "4 von 5" vorhanden sind, wuerde die KI dauernd wechseln ?
Wenn ja: das waere nur dann schlimm, wenn sie fuer den Wechsel "viel Zeit" braeuchte bzw dauernd ins Buero rennt..
Wenn nein: dann ist auch alles in Ordnung :)

Deswegen sollte die im Zweifel die "Penalty" mit beachten ...
Wenn wir nur die "Acuteness" heranziehen, kann es in allen Faellen "Gleichstand" hervorrufen (hoehere Strafe, dafuer nur noch 1 statt 2 Werbebloecke noetig ... usw.). Die KI muss also selbst "schauen", dass es unter den "noetigsten" Werbevertraegen eine eigene Ordnung reinbringt (zur Not einfach.. alphabetisch)

Reply:
Eigentlich sollte sie nicht wechseln. Wenn die Verträge "gleich" sind und beiden 3 von 5 Spots gesendet wurden, entscheidet sich die KI zufällig für einen von beiden. Für den nächsten Block wäre dann der Divisor einmal 1-0.6 und einmal 1-0.8, d.h. der zufällig gewählte Vertrag würde den Vorzug erhalten. Das reicht mir erstmal. Ein schlimmeres Problem ist ja eher, dass die KI überhaupt zwei so ähnliche Verträge (mit vielen Ausstrahlungen) hat.

Mit der gewollten Änderung soll ja insb. erreicht werden, dass ein Vertrag erfüllt wird, wenn es möglich ist. Es bringt ja nichts, einen nicht zu erfüllen um den dann stattdessen ausgewählten auch nicht zu erfüllen.